### PR TITLE
docs(concepts): Add documentation for Execution Listeners

### PR DESCRIPTION
**Pull request by ce-dmelnych**
## Description

Relates to: https://github.com/camunda/camunda/issues/16219

This PR adds documentation for Execution Listeners (EL) in Camunda 8, providing developers with a guide on how to implement and utilize ELs effectively in their workflows. 

It includes the following sections:
- **Introduction**: Overview, and use cases of ELs.
- **Getting Started**: Steps to enable ELs in BPMN models.
- **Limitations**: ELs limitations.
- **Usage**: How to define Job Workers to process EL jobs.
- **Variables**: Handling variables within ELs, including scope and dynamic variable handling.
- **References**: Links to additional documentation.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by ce-dmelnych** [comment]
> @ce-dmelnych Is this ready for review or not quite yet?

Hi @christinaausley ,

I have made some corrections, and the PR is now ready for review. However, it also requires an engineering review, so there might be some additional changes after that.

@saig0, could you please assist with the engineering review?

Thanks!

**Comment by christinaausley** [comment]
@ce-dmelnych Cool 👍 I'll have a look after @saig0.

File: docs/components/concepts/execution-listeners.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+  <ul>
+    <li><em>Apply Output Mappings</em> (if applicable): Output variables are mapped from the element's local scope to the global scope.</li>
+    <li><em>Unsubscribe from Events</em> (if applicable): The element unsubscribes from any events it was listening for.</li>
+  </ul>
+  <h4>End Execution Listeners:</h4>
+  <p>During the element completion stage, if <code>end</code> ELs are defined, they are executed after applying output mappings (if any defined). These listeners can perform cleanup tasks and have access to output variables.</p>
+  <h4>Finalize Completion:</h4>
+  <p>The element executes any additional actions required for element completion and execution transits to the next element.</p>
+  <p>By placing custom logic at strategic points within the element lifecycle, you can enhance process control, improve data handling, and integrate seamlessly with external systems.</p>
+</details>
```
Comment (saig0) [inline docs/components/concepts/execution-listeners.md:333]: :wrench: This part is relevant to use EL correctly. We should not hide it.  

:wrench: We should make this part less technical and easier to understand for users.

File: docs/components/concepts/execution-listeners.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+### Common questions
+
+- **Can execution listeners be used with all BPMN elements?**
+  - No, some elements have limitations due to their processing nature. See the table of [supported elements](#supported-elements).
+
+### Troubleshooting
+
+- **Execution Listener not triggering**
+  - Verify the configuration and ensure the element supports the specified listener type.
```
Comment (saig0) [inline docs/components/concepts/execution-listeners.md:345]: :wrench: No relevant information.

File: docs/components/concepts/execution-listeners.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+Variables set during the completion of an EL job are local to the element where the EL is defined.
+This design choice helps to prevent global variable scope cluttering, ensuring that the variables
+are contained within the context of the specific element they are associated with.
+
+### Dynamic variable handling
+
+Execution Listeners enable the dynamic fetching and calculation of variables for the next steps in the process,
+based on the current data. This functionality is particularly useful for elements such as gateways and events,
+allowing for more informed decision-making and precise control over process flow.
```
Comment (saig0) [inline docs/components/concepts/execution-listeners.md:254]: :wrench: The usage of ELs and its technical base is most relevant for the user. We should move his part to the top.

:wrench: We should improve the wording to make it easier for users to understand.

**Review by saig0 — CHANGES_REQUESTED** [review]
@ce-dmelnych thank you for writing the docs. It's a good start. :+1: 

However, I see the potential to make it more effective for users by rephrasing and restructuring some parts. I will schedule a meeting with you to discuss the improvements.

**Comment by saig0** [comment]
@ce-dmelnych I pushed the changes from our review meeting today. :rocket:

**Review by christinaausley — APPROVED** [review]
This is so clean and well documented! Great work 🚀 

Should we also add this to the [glossary](https://docs.camunda.io/docs/reference/glossary/)?

**Comment by ce-dmelnych** [comment]
Thank you both for your valuable reviews and the improvements you committed to the documentation! 🙌

@saig0 , I appreciate the meeting and the suggestions you provided, which have greatly enhanced the clarity and effectiveness of the content 🙇‍♂️ 

@christinaausley, thank you for your positive feedback!
I’ve just pushed another commit with some small improvements to the wording and formatting, fixed a broken link, and added the `Execution listener` term to the [glossary](https://preview.docs.camunda.cloud/pr-4017/docs/next/reference/glossary/#execution-listener). Please review it to see if it aligns with your expectations.

Thanks again for your support and collaboration! 🚀

**Review by christinaausley — APPROVED** [review]

**Comment by megglos** [comment]
@saig0 @ce-dmelnych  do you consider it doable to get this merged by Tuesday next week? Will include a link in the release blogpost then.

**Comment by saig0** [comment]
@megglos yes. I'll review it on Monday.

**Review by saig0 — APPROVED** [review]
LGTM. :rocket:

**Pull request merged**